### ISSUE TITLE
Gridsome fails when using the source drupal plugin 

### ISSUE THREAD

**Issue by dasjo** [gridsome/gridsome#869]
## Description

The gridsome develop step fails with the following error

Error: Type with name "DrupalTaxonomyVocabulary" does not exists

### Steps to reproduce

Install a drupal 8 site, enable json api
follow gridsome drupal steps from https://gridsome.org/plugins/@gridsome/source-drupal

dasjo@jos-macbook-pro my-gridsome-site % gridsome develop

dasjo@jos-macbook-pro my-gridsome-site % gridsome develop
Gridsome v0.7.11

Initializing plugins...
DrupalMenuLinkContent has no entities
DrupalImageStyle has no entities
DrupalCommentType has no entities
DrupalBlockContentBasic has no entities
DrupalComment has no entities
DrupalNodePage > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodePage > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodePage > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodePage > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodePage > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalNodeExperience > Failed to add node: Duplicate key for property path: [object Object]
DrupalBlockContentType has no entities
DrupalTaxonomyVocabulary has no entities
Load sources - 0.16s
Error: Type with name "DrupalTaxonomyVocabulary" does not exists
    at SchemaComposer.get (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/graphql-compose/lib/TypeStorage.js:44:13)
    at createRefType (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/graphql/createFieldTypes.js:114:39)
    at createFieldType (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/graphql/createFieldTypes.js:65:11)
    at createObjectType (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/graphql/createFieldTypes.js:76:18)
    at createFieldType (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/graphql/createFieldTypes.js:66:11)
    at createFieldTypes (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/graphql/createFieldTypes.js:13:20)
    at createFields (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/graphql/nodes/index.js:118:22)
    at createNodesSchema (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/graphql/nodes/index.js:37:5)
    at createSchema (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/graphql/createSchema.js:59:3)
    at Schema.buildSchema (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/app/Schema.js:28:28)
    at Plugins.createSchema (/Users/dasjo/Drive/04_Projects/my-gridsome-site/node_modules/gridsome/lib/app/Plugins.js:94:22)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

### Expected result

I assume the build step should work because I didn't apply any special configuration.

Am I missing something?

### Actual result

Error as described above

### Environment

  System:
    OS: macOS 10.15.1
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
  Binaries:
    Node: 12.12.0 - /usr/local/bin/node
    npm: 6.11.3 - /usr/local/bin/npm
  Browsers:
    Chrome: 78.0.3904.108
    Firefox: 70.0.1
    Safari: 13.0.3
  npmPackages:
    @gridsome/source-drupal: ^0.4.1 => 0.4.1 
    gridsome: ^0.7.0 => 0.7.11 
  npmGlobalPackages:
    @gridsome/cli: 0.3.1

Thanks for your help!

**Comment by Al-Rozhkov** [comment]
I've been using `source-drupal` plugin for a long time without any problem. I can only guess but looks like little misconfiguration. First check that you didn't set `DrupalTaxonomyVocabulary` in `templates` property of your `gridsome.config.js`.

I can say more if I see your `gridsome.config.js`.

**Comment by dasjo** [comment]
@Al-Rozhkov thanks for your reply!

so far, i haven't set anything special i think. my `gridsome.config.js` looks like the following

```
// This is where project configuration and plugin options are located. 
// Learn more: https://gridsome.org/docs/config

// Changes here require a server restart.
// To restart press CTRL + C in terminal and run `gridsome develop`

// default exclude can be imported
const { defaultExcludes } = require('@gridsome/source-drupal')

module.exports = {
  siteName: 'Gridsome',
  plugins: [
    {
      use: '@gridsome/source-drupal',
      options: {
        baseUrl: 'http://dasjo-data.docksal/',
        exclude: [ ...defaultExcludes ], // include the defaults
      }
    }
  ]
}
```

also attaching my json api export as maybe this is causing troubles?

[jsonapi.txt](https://github.com/gridsome/gridsome/files/3965255/jsonapi.txt)

thanks for you help, regards,
josef

**Comment by dasjo** [comment]
in the meantime i could confirm that pointing to another jsonapi endpoint solves the problem for me, so probably has something to do with the data / structure of this endpoint

**Comment by dasjo** [comment]
i was able to narrow down a problem a bit further.

as soon as i add a term to the vocabulary tags, the error appears.

when the vocabulary tags is empty, gridsome compiles without an issue

**Comment by dasjo** [comment]
was able to fix the issue by allowing access to the vocabularies

**Comment by bastianhilton** [comment]
for those that are a bit confused i had the issue as well but managed to fix it using Gridsome and Drupal 8.8.3 by following the below steps:

Admin Area Toolbar > People > Permissions > Allowing public (anonymous) access to vocabularies.
